app/src/main/java/fr/free/nrw/commons/media/MediaDetailPagerFragment.java

```diff
@@ -164,13 +168,19 @@ private void downloadMedia(Media m) {
         req.allowScanningByMediaScanner();
         req.setNotificationVisibility(DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED);
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !(ContextCompat.checkSelfPermission(getContext(), READ_EXTERNAL_STORAGE) == PERMISSION_GRANTED)) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
+                ContextCompat.checkSelfPermission(getContext(), READ_EXTERNAL_STORAGE)
+                        != PERMISSION_GRANTED
+                && getView() != null) {
             Snackbar.make(getView(), R.string.read_storage_permission_rationale,
```

Comment:
sorry but why? isn't it more readable to do a `something!=somethingElse` than a !(something == somethingElse)?

Reply:
I agree using !=, but with parenthesis. Because there is 5 logical operations here and parenthesis can guide us about priorities on first sight. So I meant `(something!=somethingElese)` especially in the middle. But not a very big deal, we can use as is also.